### PR TITLE
AssertJ Core 3.13.0 Release notes

### PR DESCRIPTION
**Pull request by joel-costigliola**
@scordio if you could review the release notes describing `asInstanceOf` that would be fantastic!

### CODE REVIEW

File: src/docs/asciidoc/user-guide/index.adoc [hunk 1]
```diff
@@ -28,6 +28,112 @@ include::contributors.adoc[]
...
+// Given a String declared as an Object
+Object value = "Once upon a time in the west";
+
+// With asInstanceOf, we switch to specific String assertion by specifying with the InstanceOfAssertFactory for String
+assertThat(value).asInstanceOf(STRING).startsWith("Once");
+----
+
+AssertJ verifies that the actual value is compatible with the assertions `InstanceOfAssertFactory` is going to give access to.
+
+`InstanceOfAssertFactories` provides static factories for all types AssertJ provides assertions for, additional factories can be created with custom `InstanceOfAssertFactory` instances.
```
[the remark this answers is not in the document]
Comment (joel-costigliola) [reply src/docs/asciidoc/user-guide/index.adoc]: awesome thanks!

**Comment by joel-costigliola** [comment]
Pushed some improvements, thanks for the feedback.

**Comment by scordio** [comment]
👍

File: src/docs/asciidoc/user-guide/index.adoc [hunk 2]
```diff
@@ -146,6 +161,50 @@ assertThat(hobbits).asInstanceOf(InstanceOfAssertFactories.LIST)
...
+assertThat(frodo).extracting("name")
+                 .startsWith("Fro");
+
+// To get String assertions use asInstanceOf:
+assertThat(frodo).extracting("name")
+                 .asInstanceOf(InstanceOfAssertFactories.STRING)
+                 .startsWith("Fro");
+----
+
+If the object under test is a Map, the field/property is used as a key to the map.
```
Comment (scordio) [inline src/docs/asciidoc/user-guide/index.adoc]: > the field/property is used

Minor: what about "the parameter is used"?

File: src/docs/asciidoc/user-guide/assertions-guide.adoc [hunk 3]
```diff
@@ -264,12 +264,99 @@ assertThat(actual).isEqualTo(expected).usingComparator(new CustomComparator());
...
+
+===== MaxElementsForPrinting
+
+===== MaxLengthForSingleLineDescription
+
+===== RemoveAssertJRelatedElementsFromStackTrace
+
+===== Representation
+
+===== isterCustomDateFormat
```
Comment (scordio) [inline src/docs/asciidoc/user-guide/assertions-guide.adoc]: Typo `RegisterCustomDateFormat`

Comment (joel-costigliola) [reply src/docs/asciidoc/user-guide/assertions-guide.adoc]: good catch!

**Pull request merged**